src/java/org/apache/cassandra/db/ColumnFamilyStore.java

```diff
@@ -1538,6 +1538,14 @@ public CompactionManager.AllSSTableOpStatus relocateSSTables(int jobs) throws Ex
 
     public CompactionManager.AllSSTableOpStatus garbageCollect(TombstoneOption tombstoneOption, int jobs) throws ExecutionException, InterruptedException
     {
+        if (memtableWritesAreDurable())
```

Comment:
Semantically this is not the right test (the memtable may be durable but still flush normally e.g. by having an internal commit log); also, it may be beneficial for a memtable to initiate a flush in response to this.

Let's leave this decision (and the `isClean` check below) to the memtable itself.